### PR TITLE
[Refactoring] Little refactoring on JMAP contracts

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/BackReferenceContract.scala [hunk 1]
```diff
@@ -43,6 +44,7 @@ trait BackReferenceContract {
   }
 
   @Test
+  @Tag(CategoryTags.BASIC_FEATURE)
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/BackReferenceContract.scala:47]: No that's unecessary IMO as backReference do not depend of any backend.

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 2]
```diff
@@ -3590,6 +3597,7 @@ trait MailboxSetMethodContract {
   }
 
   @Test
+  @Tag(CategoryTags.BASIC_FEATURE)
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:3600]: IMO that one is not necessary

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 3]
```diff
@@ -5517,6 +5528,7 @@ trait MailboxSetMethodContract {
...
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:5531]: Remember: it is smoke tests. The ratio should be between 15% to 20% per test class.

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala [hunk 4]
```diff
@@ -218,7 +220,8 @@ trait VacationResponseSetMethodContract {
   }
 
   @Test
-  def partialUpdateShouldSucceed(server: GuiceJamesServer): Unit = {
+  @Tag(CategoryTags.BASIC_FEATURE)
+  def partialUpdateShouldSucceed(): Unit = {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala:224]: I think this one is not needed

**Review by remk — APPROVED** [review]

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala (hunk 3 above)
Comment (Arsnael) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:5531]: Yes... 11 smoke tests for a total of 111 tests (yes there is this much for Mailbox/set)... makes something around like 10% I believe? :trollface:

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**